raft/raftpb/raft.proto

```diff
@@ -75,14 +75,26 @@ message HardState {
 	optional uint64 commit = 3 [(gogoproto.nullable) = false];
 }
 
+enum SuffrageState {
+	Nonvoter = 0;
+	Voter    = 1;
+}
+
+message Server {
+	optional uint64        node     = 1 [(gogoproto.nullable) = false];
+	optional SuffrageState suffrage = 2 [(gogoproto.nullable) = false];
+}
+
 message ConfState {
-	repeated uint64 nodes = 1;
+	repeated uint64 nodes   = 1;
+	repeated Server servers = 2;
```

Comment:
can we make this simpler by just having

```
repeated uint64 nodes = 1;
repeated uint64 learners= 2;
```
?

Reply:
we may also have staging nodes after addVoter feature

Reply:
i do not like the idea of staging server. i would rather let the application itself promote the leader to a node explicitly.

Reply:
basically, the less raft layer needs to care about the better.